index.bs

```diff
@@ -2837,166 +2843,199 @@ visible through the {{WritableStream}}'s public API.
   1. Set _stream_.[[storedError]] to _reason_.
-  1. If _wasAborted_ is *false* and _stream_.[[writer]] is not *undefined*, perform !
-     WritableStreamDefaultWriterEnsureReadyPromiseRejectedWith(_stream_.[[writer]], _reason_, _readyPromiseIsPending_).
-  1. Perform ! WritableStreamRejectPromisesInReactionToError(_stream_).
+  1. If _stream_.[[pendingAbortRequest]] is *undefined*,
+    1. Let _writer_ be _stream_.[[writer]].
+    1. If _writer_ is not *undefined*,
+      1. If ! WritableStreamCloseQueuedOrInFlight(_stream_) is *false* and _stream_.[[backpressure]] is *true*,
+         <a>reject</a> _writer_.[[readyPromise]] with _reason_.
+      1. Otherwise, set _writer_.[[readyPromise]] to <a>a promise rejected with</a> _reason_.
+      1. Set _writer_.[[readyPromise]].[[PromiseIsHandled]] to *true*.
+  1. Otherwise,
+    1. <a>Reject</a> _stream_.[[pendingAbortRequest]] with _reason_.
+    1. Set _stream_.[[pendingAbortRequest]] to *undefined*.
+  1. Perform ! WritableStreamRejectPromisesInReactionToError(_stream_, _reason_).
 </emu-alg>
 
-<h4 id="writable-stream-finish-pending-close" aoid="WritableStreamFinishPendingClose"
-nothrow>WritableStreamFinishPendingClose ( <var>stream</var> )</h4>
+<h4 id="writable-stream-finish-in-flight-close" aoid="WritableStreamFinishInFlightClose"
+nothrow>WritableStreamFinishInFlightClose ( <var>stream</var> )</h4>
 
 <emu-alg>
-  1. Assert: _stream_.[[pendingCloseRequest]] is not *undefined*.
-  1. <a>Resolve</a> _stream_.[[pendingCloseRequest]] with *undefined*.
-  1. Set _stream_.[[pendingCloseRequest]] to *undefined*.
+  1. Assert: _stream_.[[inFlightCloseRequest]] is not *undefined*.
+  1. <a>Resolve</a> _stream_.[[inFlightCloseRequest]] with *undefined*.
+  1. Set _stream_.[[inFlightCloseRequest]] to *undefined*.
   1. Let _state_ be _stream_.[[state]].
-  1. Let _wasAborted_ be *false*.
-  1. If _stream_.[[pendingAbortRequest]] is not *undefined*, set _wasAborted_ to *true*.
   1. If _state_ is `"errored"`,
-    1. if _wasAborted_ is *true*,
-      1. <a>Reject</a>  _stream_.[[pendingAbortRequest]] with _stream_.[[storedError]].
-      1. Set _stream_.[[pendingAbortRequest]] to *undefined*.
-    1. Perform ! WritableStreamRejectClosedPromiseIfAny(_stream_).
+    1. Perform ! WritableStreamFinishInFlightCloseInErroredState(_stream_).
     1. Return.
-  1. Assert: _state_ is `"closing"`.
-  1. If _wasAborted_ is *false*,
+  1. Assert: _state_ is `"writable"`.
+  1. If _stream_.[[pendingAbortRequest]] is *undefined*,
+    1. Set _stream_.[[state]] to `"closed"`.
     1. Let _writer_ be _stream_.[[writer]].
     1. If _writer_ is not *undefined*, <a>resolve</a> _writer_.[[closedPromise]] with *undefined*.
-    1. Set _stream_.[[state]] to `"closed"`.
     1. Return.
   1. <a>Resolve</a> _stream_.[[pendingAbortRequest]] with *undefined*.
   1. Set _stream_.[[pendingAbortRequest]] to *undefined*.
+  1. Let _error_ to a new *TypeError* indicating that the stream was closed successfully after abort request.
```

Comment:
I was struggling with this since
- it should represent the situation of the stream. not the abort operation
- it should tell what kind of bad thing has happened and how it was bad for the stream

Since the stream has finished closing successfully, the only bad things we should talk about are:
- the fact that there was an attempt to abort the stream (which didn't take place eventually)
- that the pending abort has been resolved without running sink.abort().